src/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.java

```diff
@@ -0,0 +1,142 @@
+package com.airbnb.android.react.lottie;
+
+import android.util.Log;
+import android.widget.ImageView;
+
+import com.airbnb.lottie.LottieAnimationView;
+
+import org.json.JSONObject;
+
+import java.lang.ref.WeakReference;
+
+public class LottieAnimationViewPropertyManager {
+
+  private final WeakReference<LottieAnimationView> viewWeakReference;
+
+  private JSONObject animationJson;
+  private Float progress;
+  private Boolean loop;
+  private Float speed;
+
+  /**
+   * Should be set to true if one of the animationName related parameters has changed as result of latest
+   * reconciliation. We need to update the animation in this case.
+   */
+  private boolean animationNameDirty;
+
+  private String animationName;
+  private LottieAnimationView.CacheStrategy cacheStrategy;
+  private Boolean useHardwareAcceleration;
+  private ImageView.ScaleType scaleType;
+  private String imageAssetsFolder;
+  private Boolean enableMergePaths;
+
+  public LottieAnimationViewPropertyManager(LottieAnimationView view) {
+    this.viewWeakReference = new WeakReference<>(view);
+  }
+
+  public void setAnimationName(String animationName) {
+    this.animationName = animationName;
+    this.animationNameDirty = true;
+  }
+
+  public void setAnimationJson(String json) {
+    try {
+      this.animationJson = new JSONObject(json);
+    } catch (Exception e) {
+       // TODO: expose this to the user better. maybe an `onError` event?
+       Log.e(LottieAnimationViewManager.TAG,"setSourceJsonError", e);
+    }
+  }
+
+  public void setCacheStrategy(LottieAnimationView.CacheStrategy strategy) {
+    this.cacheStrategy = strategy;
+    this.animationNameDirty = true;
+  }
+
+  public void setProgress(Float progress) {
+    this.progress = progress;
+  }
+
+  public void setSpeed(float speed) {
+    this.speed = speed;
+  }
+
+  public void setLoop(boolean loop) {
+    this.loop = loop;
+  }
+
+  public void setUseHardwareAcceleration(boolean useHardwareAcceleration) {
+    this.useHardwareAcceleration = useHardwareAcceleration;
+  }
+
+  public void setScaleType(ImageView.ScaleType scaleType) {
+    this.scaleType = scaleType;
+  }
+
+  public void setImageAssetsFolder(String imageAssetsFolder) {
+    this.imageAssetsFolder = imageAssetsFolder;
+  }
+
+  public void setEnableMergePaths(boolean enableMergePaths) {
+    this.enableMergePaths = enableMergePaths;
+  }
+
+  /**
+   * Updates the view with changed fields.
+   * Majority of the properties here are indepenbent so they are has to be reset to null
+   * as soon as view is updated with the value.
+   *
+   * The only exception from this rule is the group of the properties for the animation.
+   * For now this is animationName and cacheStrategy. These two properties are should be set
+   * simultaneously if the dirty flag is set.
+   */
+  public void commitChanges() {
+    LottieAnimationView view = viewWeakReference.get();
+    if (view != null) {
```

Comment:
Just return early if view == null so the entire method doesn't have to me nested